homeassistant/components/plaato/__init__.py

```diff
@@ -0,0 +1,125 @@
+"""Support for Plaato Airlock."""
+import logging
+
+from aiohttp import web
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN as SENSOR
+from homeassistant.const import (
+    CONF_WEBHOOK_ID, HTTP_OK, HTTP_UNPROCESSABLE_ENTITY,
+    TEMP_CELSIUS, TEMP_FAHRENHEIT, VOLUME_GALLONS, VOLUME_LITERS)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['webhook']
+
+PLAATO_DEVICE_SENSORS = 'sensors'
+PLAATO_DEVICE_ATTRS = 'attrs'
+
+ATTR_DEVICE_ID = 'device_id'
+ATTR_DEVICE_NAME = 'device_name'
+ATTR_TEMP_UNIT = 'temp_unit'
+ATTR_VOLUME_UNIT = 'volume_unit'
+ATTR_BPM = 'bpm'
+ATTR_TEMP = 'temp'
+ATTR_SG = 'sg'
+ATTR_OG = 'og'
+ATTR_BUBBLES = 'bubbles'
+ATTR_ABV = 'abv'
+ATTR_CO2_VOLUME = 'co2_volume'
+ATTR_BATCH_VOLUME = 'batch_volume'
+
+SENSOR_UPDATE = '{}_sensor_update'.format(DOMAIN)
+
+WEBHOOK_SCHEMA = vol.Schema({
+    vol.Required(ATTR_DEVICE_NAME): cv.string,
+    vol.Required(ATTR_DEVICE_ID): cv.positive_int,
+    vol.Required(ATTR_TEMP_UNIT): vol.Any(TEMP_CELSIUS, TEMP_FAHRENHEIT),
+    vol.Required(ATTR_VOLUME_UNIT): vol.Any(VOLUME_LITERS, VOLUME_GALLONS),
+    vol.Required(ATTR_BPM): cv.positive_int,
+    vol.Required(ATTR_TEMP): vol.Coerce(float),
+    vol.Required(ATTR_SG): vol.Coerce(float),
+    vol.Required(ATTR_OG): vol.Coerce(float),
+    vol.Required(ATTR_ABV): vol.Coerce(float),
+    vol.Required(ATTR_CO2_VOLUME): vol.Coerce(float),
+    vol.Required(ATTR_BATCH_VOLUME): vol.Coerce(float),
+    vol.Required(ATTR_BUBBLES): cv.positive_int,
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, hass_config):
+    """Set up the Plaato component."""
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Configure based on config entry."""
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    webhook_id = entry.data[CONF_WEBHOOK_ID]
+    hass.components.webhook.async_register(
+        DOMAIN, 'Plaato', webhook_id, handle_webhook)
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, SENSOR)
+    )
+
+    return True
+
+
+async def async_unload_entry(hass, entry):
```

Comment:
You also need to call the dispatcher unsubscribe function stored in `hass.data["plaato.sensor"]`

Reply:
@balloob I'm not sure what you mean. Can you refer to some documentation regarding this?

The only thing I could find in other components was something like this
```python
hass.data[DOMAIN].pop(entry.entry_id)
```

Reply:
@balloob I don't understand what you mean. Can yo refer to documentation or an example?

Reply:
You connect a dispatcher when you set up the entry. So during unloading, you need to disconnect that dispatcher signal listener.

Reply:
Done!